iocore/net/quic/test/test_QUICHandshakeProtocol.cc

```diff
@@ -483,23 +448,16 @@ TEST_CASE("QUICHandshakeProtocol")
 
     REQUIRE(client->handshake(&msg3, &msg2_1) == 1);
     REQUIRE(client->handshake(&msg3, &msg2_2) == 1);
-#else
-    REQUIRE(client->handshake(&msg3, &msg2) == 1);
-#endif
 
     // NS
-    QUICHandshakeMsgs msg4;
-    uint8_t msg4_buf[MAX_HANDSHAKE_MSG_LEN] = {0};
-    msg4.buf                                = msg4_buf;
-    msg4.max_buf_len                        = MAX_HANDSHAKE_MSG_LEN;
-
-    REQUIRE(server->handshake(&msg4, &msg3) == 1);
-
-    QUICHandshakeMsgs msg5;
-    uint8_t msg5_buf[MAX_HANDSHAKE_MSG_LEN] = {0};
-    msg5.buf                                = msg5_buf;
-    msg5.max_buf_len                        = MAX_HANDSHAKE_MSG_LEN;
-    REQUIRE(client->handshake(&msg5, &msg4) == 1);
+    QUICHandshakeMsgs *msg4 = nullptr;
+    REQUIRE(server->handshake(&msg4, msg3) == 1);
+    REQUIRE(msg4);
+
+    // This doesn't work with BoringSSL (what's expected?)
+    // QUICHandshakeMsgs *msg5 = nullptr;
```

Comment:
@masaori335 Can you recall what message this block expects?

Reply:
I completely forgot.
But, assuming `msg4` is New Session Ticket from the server, it looks like `REQUIRE(client->handshake(&msg5, &msg4) == 1);` is checking client could accept New Session Ticket.
What happens with BoringSSL?

Reply:
I changed `handshake()` interface a little as you can see. The first parameter `out` will be non-null only if there's something has to be sent. And it was null here.

It sounds like we should check `client->handshake() == 1` and `msg5 == nullptr` because there is no expected data to send. Does it make sense?

Reply:
Make sense.